src/components/AssetsComponent.jsx

```diff
@@ -63,10 +63,16 @@ export default class AssetsComponent extends Component {
     const { status } = match.params;
 
     this.props.setActivePage(activePage);
-    const currentPageList = this.props.assetsList[`page_${activePage}`];
+    const filters = this.props.selected;
 
-    if (isEmpty(currentPageList)) {
-      this.retrieveAssets(activePage, this.state.limit, status);
+    if (filters) {
+      this.props.getAssetsAction(activePage, this.state.limit, filters);

```

Comment:
I don't recommend doing it this way @rotimi-babalola. The way we have implemented pagination has a tradeoff between performance and memory. That is the reason we have `const CUTOFF_LIMIT = 50;` at the top of the file. Once the cutoff limit is achieved, any request for more assets will be fetched from the API all the time. Therefore, instead of implementing it this way, I recommend that you do it as follows:

1. Update the cutoff limit to 100. This is not related to the scope of this bug but it is a small change to make, therefore I slotted it in 😄 
```jsx
const CUTOFF_LIMIT = 100;
```

2. Revert to how `handlePaginationChange` was before you updated it. After which, pass the filters to `retrieveAssets(...)` as the last parameter. I.e.
```jsx
handlePaginationChange = (e, { activePage }) => {
  const { match, selected } = this.props;
  ...
  if (isEmpty(currentPageList)) {
    this.retrieveAssets(
      activePage,
      this.state.limit,
      status,
      selected
     );
  }
}
```
3. Update `retrieveAssets` to use the filters. Also, instead of creating the URL directly in it, import `constructUrl` from `_utils/assets.js` and use it. I.e.
```jsx
retrieveAssets = (activePage, limit, status, filters = {}) => {
  if (checkIfCutoffExceeded(activePage, limit)) {
    const url  = constructUrl(activePage, limit, filters, status);
    ....
  }

  return this.loadAssets(activePage, limit, filters, status);
}
```